cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -485,15 +486,8 @@ private String metadataKeyFromColumnKey(byte[] columnKey) {
   /**
    * Generate an output path for a given partition key.
    */
-  // package visible for PartitionedFileSetDefinition
-  static String getOutputPath(Partitioning partitioning, PartitionKey key) {
-    StringBuilder builder = new StringBuilder();
-    String sep = "";
-    for (String fieldName : partitioning.getFields().keySet()) {
-      builder.append(sep).append(key.getField(fieldName).toString());
-      sep = "/";
-    }
-    return builder.toString();
+  public static String getOutputPath(PartitionKey key) {
```

Comment:
so, previously this would fail if the partition key is incomplete. (not with a good message, though). Now it simply uses the given key to construct a path. Also, previously, the order of the fields in the path was given by the partitioning (fixed and always he same). Now it depends on the key (which is not guaranteed to be in the same order).